Core/include/Acts/Vertexing/TrackLinearizer.hpp

```diff
@@ -0,0 +1,25 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/EventData/TrackParameters.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/MagneticField/MagneticFieldContext.hpp"
+#include "Acts/MagneticField/MagneticFieldProvider.hpp"
+#include "Acts/Utilities/Delegate.hpp"
+
+namespace Acts {
+struct LinearizedTrack;
+class Surface;
+
+using TrackLinearizer = Acts::Delegate<Result<LinearizedTrack>(
+    const BoundTrackParameters&, double, const Surface& perigeeSurface,
+    const Acts::GeometryContext&, const Acts::MagneticFieldContext&,
+    MagneticFieldProvider::Cache&)>;
```

Comment:
```suggestion
    const BoundTrackParameters& parameters, double, const Surface& perigeeSurface,
    const Acts::GeometryContext& geoContext, const Acts::MagneticFieldContext& magContext,
    MagneticFieldProvider::Cache& magProvider)>;
```

Reply:
I did not know you could do that, actually.

Reply:
I'll model it after the argument names in `HelicalTrackLinearizer` now.